src/utils/SupabaseActions.ts

```diff
@@ -43,6 +43,25 @@ export class SupabaseActions extends SupabaseService {
     return filePayload
   }
 
+  async getFilePathFromUrl(url: string) {
+    try {
+      if (url) {
+        const parsedUrl = new URL(url)
+        const pathname = parsedUrl.pathname
+        const mediaIndex = pathname.indexOf('/media/')
+        if (mediaIndex !== -1) {
+          const filePath = pathname.substring(mediaIndex + '/media/'.length)
+          return filePath
+        } else {
+          return null
+        }
```

Comment:
```suggestion
        const filePath = pathname.split('/media/')[1]
```

I think this should work. Can you check?

Reply:
Also, I think if condition can be omitted if you do URL(url ?? '') or maybe even that is not needed.